esp/bindings/http/platform/httpservice.cpp

```diff
@@ -1120,6 +1125,208 @@ EspAuthState CEspHttpServer::preCheckAuth(EspAuthRequest& authReq)
+{
+    const char* name = cookie.cookieName.get();
+    if (strieq(name, "ESPSessionID") || strieq(name, authReq.authBinding->querySessionIDCookieName()))
+    {
+        if (verifyESPSessionIDCookie(authReq))
+            cookie.valid.set("true");
+    }
+    else if (strieq(name, SESSION_AUTH_OK_COOKIE))
+        verifyESPAuthenticatedCookie(authReq, cookie);
+    else if (strieq(name, USER_NAME_COOKIE))
+        verifyESPUserNameCookie(authReq, cookie);
+    else if (strieq(name, SESSION_TIMEOUT_COOKIE) || strieq(name, SESSION_AUTH_MSG_COOKIE))
+    {
+        //SESSION_TIMEOUT_COOKIE: used to pass timeout settings to a client.
+        //SESSION_AUTH_MSG_COOKIE: used to pass authentication message to a client.
+        //A client should clean it as soon as received. ESP always returns invalid if it is asked.
+        cookie.verificationDetails.set("ESP cannot verify this cookie. It is one-time use only.");
+    }
+    else if (strieq(name, SESSION_START_URL_COOKIE))
+    {   
+        //SESSION_START_URL_COOKIE: created as soon as a login process is started and used to redirect at the end of the login process.
+        //After the login process, ESP should remove this cookie if possible. This verify function is designed to verify cookies before 
+        //a login process is started. At that time, any SESSION_START_URL_COOKIE should be invalid.
+        cookie.verificationDetails.set("ESP cannot verify this cookie. This cookie is only valid within a login process.");
+    }
+    else if (strieq(name, SESSION_ID_TEMP_COOKIE))
+    {   
+        //SESSION_ID_TEMP_COOKIE: used for remembering UserID when a user is in AS_PASSWORD_VALID_BUT_EXPIRED state and updating password.
+        //It is created before the updatepasswordinput is started. After the update process, ESP should remove this cookie if possible.
+        //This verify function is designed to verify cookies before a login process is started. At that time, any SESSION_ID_TEMP_COOKIE
+        //should be invalid.
+        cookie.verificationDetails.set("ESP cannot verify this cookie. This cookie is only valid when updating an expired passwsord.");
+    }
+    else
+        cookie.verificationDetails.set("ESP does not know this cookie");
+}
+
+void CEspHttpServer::verifyESPUserNameCookie(EspAuthRequest& authReq, CESPCookieVerification& cookie)
+{
+    //USER_NAME_COOKIE: used in UserNameOnlyMode for remembering the user name. Right now it is
+    //always valid if ESP is in the UserNameOnlyMode.
+    if (authReq.authBinding->getDomainAuthType() != AuthUserNameOnly)
+        return;
+
+    if (!cookie.cookieValue.get())
+        cookie.cookieValue.set(m_request->queryCookie(USER_NAME_COOKIE));
+    if (cookie.cookieValue.get())
+        cookie.valid.set("true");
+}
+
+bool CEspHttpServer::verifyESPSessionIDCookie(EspAuthRequest& authReq)
+{
+    unsigned sessionID = readCookie(authReq.authBinding->querySessionIDCookieName());
+    if (sessionID == 0)
+        return false;
```

Comment:
When could sessionID be 0?  No sessions enabled?  Perhaps a comment here would help